src/main/java/bio/terra/service/common/CommonFlightUtils.java

```diff
@@ -14,4 +21,37 @@ public static Instant getCreatedAt(FlightContext context) {
     }
     return null;
   }
+
+  /**
+   * Obtains flight information of Interest that might be useful in a journal entry. Patterned after
+   * an allowList because the flightContext input parameters can contain some sensitive information
+   * that we don't want included in a journal entry. The pattern of using the request payload to
+   * hold arbitrary objects that can be returned as JSON led to the inclusion of a hint in the key
+   * name so that our front end can detect and unwrap this value.
+   *
+   * @param flightContext
+   * @return LinkedHashMap<String, String>
+   */
+  public static Map<String, String> getFlightInformationOfInterest(
+      @NotNull FlightContext flightContext) {
+    Map<String, String> results = new LinkedHashMap<>();
```

Comment:
Not explicitly, but it will be serialized and consumed as a Javascript backed Map.  https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map. Key insertion order is retained.

Whenever I create a java Map that I know will be serialized as JSON, I prefer to acknowledge the consuming implementation.  Would you prefer something different here?

Reply:
Unless the API you're providing guarantees map key ordering, I don't see a reason to provide strict ordering. Having code rely on ordering makes it more fragile and (in most cases) doesn't provide a benefit. So while JSON Map type does provide an ordering when converted to a Java type, I don't think it's important to use an ordered type as an input. To assist in testing, I would use a library like JSONAssert, which is included as part of spring boot test, rather than make a test depend on key ordering.